eks/cuda-vector-add/cuda-vector-add.go

```diff
@@ -0,0 +1,231 @@
+	return &tester{cfg: cfg}
+}
+
+type tester struct {
+	cfg Config
+}
+
+func (ts *tester) AggregateResults() error {
+	if !ts.cfg.EKSConfig.IsEnabledAddOnCudaVectorAdd() {
+		ts.cfg.Logger.Info("skipping tester.AggregateResults", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+		return nil
+	}
+	if !ts.cfg.EKSConfig.AddOnCudaVectorAdd.Created {
+		ts.cfg.Logger.Info("skipping tester.AggregateResults", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+		return nil
+	}
+
+	ts.cfg.Logger.Info("starting tester.AggregateResults", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	return nil
+}
+
+func (ts *tester) Create() error {
+	if ts.cfg.EKSConfig.AddOnCudaVectorAdd.Created {
+		ts.cfg.Logger.Info("skipping create AddOnCudaVectorAdd")
+		return nil
+	}
+
+	ts.cfg.EKSConfig.AddOnCudaVectorAdd.Created = true
+	ts.cfg.EKSConfig.Sync()
+	createStart := time.Now()
+	defer func() {
+		createEnd := time.Now()
+		ts.cfg.EKSConfig.AddOnCudaVectorAdd.TimeFrameCreate = timeutil.NewTimeFrame(createStart, createEnd)
+		ts.cfg.EKSConfig.Sync()
+	}()
+
+	if err := k8s_client.CreateNamespace(
+		ts.cfg.Logger,
+		ts.cfg.K8SClient.KubernetesClientSet(),
+		ts.cfg.EKSConfig.AddOnCudaVectorAdd.Namespace,
+	); err != nil {
+		return err
+	}
+
+	ts.cfg.Logger.Info("creating pod",
+		zap.String("name", "CudaVectorAdd"),
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	_, err := ts.cfg.K8SClient.
+		KubernetesClientSet().
+		CoreV1().
+		Pods(ts.cfg.EKSConfig.AddOnCudaVectorAdd.Namespace).
+		Create(
+			ctx,
+			&v1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Pod",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "vector-add",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyOnFailure,
+					Containers: []v1.Container{
+						{
+							Name:  "cuda-vector-add",
+							// https://github.com/kubernetes/kubernetes/blob/v1.7.11/test/images/nvidia-cuda/Dockerfile
+							Image: "k8s.gcr.io/cuda-vector-add:v0.1",
+							Resources: v1.ResourceRequirements{
+								Limits: map[v1.ResourceName]resource.Quantity{
+									v1.ResourceName("nvidia.com/gpu"): resource.MustParse("1"),
+								},
+							},
+						},
+					},
+				},
+			},
+			metav1.CreateOptions{},
+		)
+	cancel()
+	if err != nil {
+		return fmt.Errorf("failed to create Pod (%v)", err)
+	}
+
+	retryStart, waitDur := time.Now(), 3*time.Minute
+
+	for time.Now().Sub(retryStart) < waitDur {
+		select {
+		case <-ts.cfg.Stopc:
+			return errors.New("cuda-vector-add pod check aborted")
+		case <-time.After(5 * time.Second):
+		}
+
+		ts.cfg.Logger.Info("querying cuda-vector-add")
+		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		out, err := ts.cfg.K8SClient.
+			KubernetesClientSet().
+			CoreV1().
+			Pods(ts.cfg.EKSConfig.AddOnCudaVectorAdd.Namespace).
+			Get(
+				ctx,
+				"vector-add",
```

Comment:
Can we define `const` for this pod name?

Reply:
Yeah good point, fixed.